File: dotnet/src/MableBank.Application/NullLogger.cs
using MableBank.Application.Telemetry;

namespace MableBank.Application;

public sealed class NullLogger : ILogger
{
    public static NullLogger Instance { get; } = new();

    private NullLogger()
    {
    }

    public void LogTransferProcessed(TransferProcessedTelemetry telemetry)
    {
        _ = telemetry;
    }
}

File: ruby/lib/mable_bank/application/null_logger.rb
# frozen_string_literal: true

module MableBank
  module Application
    module LoggerPort
      def log_transfer_processed(_telemetry)
        raise NotImplementedError, "#{self.class} must implement #log_transfer_processed"
      end
    end

    class NullLogger
      include LoggerPort

      def log_transfer_processed(_telemetry)
        nil
      end
    end
  end
end
